dvc/utils/http.py

```diff
@@ -0,0 +1,100 @@
+from contextlib import contextmanager
+import io
+
+import requests
+
+from dvc.utils.compat import is_py2
+
+
+@contextmanager
+def open_url(url, mode="r", encoding=None):
+    """Opens an url as a readable stream.
+
+    Resumes on connection error.
+    Url could be a string or a callable returning a string.
+    """
+    assert mode in {"r", "rt", "rb"}
+
+    with iter_url(url) as (response, it):
+        bytes_stream = IterStream(it)
+
+        if mode == "rb":
+            yield bytes_stream
+        else:
+            encoding = encoding or response.encoding
+            yield io.TextIOWrapper(bytes_stream, encoding=encoding)
+
+
+@contextmanager
+def iter_url(url, chunk_size=io.DEFAULT_BUFFER_SIZE):
```

Comment:
Codeclimate is not happy about you cramping full blown functions into another function. If helpers are not trivial , consider moving them outside.

Reply:
They don't make sense outside.

Reply:
BTW, code climate should add up complexity of nested functions, this is the same as summing up complexities of all the methods in a class )

Reply:
@Suor Class is not the same as a function, please split it. Again, we don't do such things in other places.

Reply:
Ok, we've discussed this privately and decided to go with this :slightly_smiling_face: